code/modules/mob/living/living_defense.dm

```diff
@@ -165,7 +165,10 @@
 					log_combat(user, src, "attempted to strangle", addition="kill grab")
 			if(!do_mob(user, src, grab_upgrade_time))
 				return 0
-			if(!user.pulling || user.pulling != src || user.grab_state != old_grab_state || user.a_intent != INTENT_GRAB)
+			if(!user.pulling || user.pulling != src || user.grab_state != old_grab_state)
+				return 0
+			if(user.a_intent != INTENT_GRAB)
+				to_chat(user, "<span class='notice'>You cannot upgrade your grab further when not on grab intent!<span>")

```

Comment:
Would probably be phrased better as something like "You must be on grab intent to upgrade your grab!"

Reply:
*upgrade your grab _further_

Reply:
```suggestion
				to_chat(user, "<span class='notice'>To upgrade your grab[user.grab_state > GRAB_PASSIVE ? " further" : ""], you must be on grab intent!<span>") // Going from not grabbing to passive grabbing isn't an "upgrade"
```

Reply:
the check already doesn't run when making passive or aggressive grabs